core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -961,11 +964,7 @@ protected final void closeNoLock(String reason) {
                 }
                 // no need to commit in this case!, we snapshot before we close the shard, so translog and all sync'ed
                 logger.trace("rollback indexWriter");
-                try {
-                    indexWriter.rollback();
-                } catch (AlreadyClosedException e) {
-                    // ignore
-                }
+                indexWriter.rollback();
```

Comment:
if we hit a merge exception in multiple threads we can call this multiple times? Why is it not ok to swallow this? do we need more assertions here or commments?